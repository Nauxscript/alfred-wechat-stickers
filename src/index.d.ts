export interface Sticker {
  key: string;
}
declare module "*.json" {
  const jsonValue: any;
  export default jsonValue;
}
