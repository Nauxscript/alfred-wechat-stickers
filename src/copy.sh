query=$1
parentDir=$(pwd)

echo -n $query
echo -n $parentDir

file-to-clipboard() {
  osascript \
    -e 'on run args' \
    -e 'set the clipboard to POSIX file (first item of args)' \
    -e 'return the clipboard' \
    -e end \
    "$@"
}

file-to-clipboard $query