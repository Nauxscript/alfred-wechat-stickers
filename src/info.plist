<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>bundleid</key>
	<string>nauxscript.workflow.alfred-wechat-stickers</string>
	<key>connections</key>
	<dict>
		<key>09857D2F-1057-4384-8BF6-74F4AD3014E2</key>
		<array>
			<dict>
				<key>destinationuid</key>
				<string>5792546E-9F7A-45BA-8D4B-911714944E49</string>
				<key>modifiers</key>
				<integer>0</integer>
				<key>modifiersubtext</key>
				<string></string>
				<key>vitoclose</key>
				<false/>
			</dict>
		</array>
		<key>1A47A645-DB97-46F0-9591-32B0C3A0DD62</key>
		<array>
			<dict>
				<key>destinationuid</key>
				<string>09857D2F-1057-4384-8BF6-74F4AD3014E2</string>
				<key>modifiers</key>
				<integer>0</integer>
				<key>modifiersubtext</key>
				<string></string>
				<key>vitoclose</key>
				<false/>
			</dict>
			<dict>
				<key>destinationuid</key>
				<string>593EE2CA-735D-4E27-946D-0B068037202E</string>
				<key>modifiers</key>
				<integer>262144</integer>
				<key>modifiersubtext</key>
				<string></string>
				<key>vitoclose</key>
				<false/>
			</dict>
		</array>
		<key>593EE2CA-735D-4E27-946D-0B068037202E</key>
		<array>
			<dict>
				<key>destinationuid</key>
				<string>5792546E-9F7A-45BA-8D4B-911714944E49</string>
				<key>modifiers</key>
				<integer>0</integer>
				<key>modifiersubtext</key>
				<string></string>
				<key>vitoclose</key>
				<false/>
			</dict>
		</array>
	</dict>
	<key>createdby</key>
	<string>Nauxscript</string>
	<key>description</key>
	<string>an Alfred workflow that makes it easy to pick / copy / input a wechat's stickers.</string>
	<key>disabled</key>
	<false/>
	<key>name</key>
	<string>Wechat Stickers Search</string>
	<key>objects</key>
	<array>
		<dict>
			<key>config</key>
			<dict>
				<key>autopaste</key>
				<true/>
				<key>clipboardtext</key>
				<string>{query}</string>
				<key>ignoredynamicplaceholders</key>
				<false/>
				<key>transient</key>
				<false/>
			</dict>
			<key>type</key>
			<string>alfred.workflow.output.clipboard</string>
			<key>uid</key>
			<string>09857D2F-1057-4384-8BF6-74F4AD3014E2</string>
			<key>version</key>
			<integer>3</integer>
		</dict>
		<dict>
			<key>config</key>
			<dict>
				<key>concurrently</key>
				<false/>
				<key>escaping</key>
				<integer>0</integer>
				<key>script</key>
				<string>
					{{update_script}}
				</string>
				<key>scriptargtype</key>
				<integer>1</integer>
				<key>scriptfile</key>
				<string></string>
				<key>type</key>
				<integer>0</integer>
			</dict>
			<key>type</key>
			<string>alfred.workflow.action.script</string>
			<key>uid</key>
			<string>5792546E-9F7A-45BA-8D4B-911714944E49</string>
			<key>version</key>
			<integer>2</integer>
		</dict>
		<dict>
			<key>config</key>
			<dict>
				<key>alfredfiltersresults</key>
				<false/>
				<key>alfredfiltersresultsmatchmode</key>
				<integer>0</integer>
				<key>argumenttreatemptyqueryasnil</key>
				<false/>
				<key>argumenttrimmode</key>
				<integer>0</integer>
				<key>argumenttype</key>
				<integer>1</integer>
				<key>escaping</key>
				<integer>102</integer>
				<key>keyword</key>
				<string>ww</string>
				<key>queuedelaycustom</key>
				<integer>3</integer>
				<key>queuedelayimmediatelyinitially</key>
				<true/>
				<key>queuedelaymode</key>
				<integer>0</integer>
				<key>queuemode</key>
				<integer>1</integer>
				<key>runningsubtext</key>
				<string></string>
				<key>script</key>
				<string>osascript -l JavaScript index.js "$1" 2&gt;&amp;1</string>
				<key>scriptargtype</key>
				<integer>1</integer>
				<key>scriptfile</key>
				<string></string>
				<key>subtext</key>
				<string>比如：doge \ 笑 \ 尴尬 等</string>
				<key>title</key>
				<string>根据你的直觉键入表情关键词 \ 拼音 \ 首字母搜索...</string>
				<key>type</key>
				<integer>0</integer>
				<key>withspace</key>
				<true/>
			</dict>
			<key>type</key>
			<string>alfred.workflow.input.scriptfilter</string>
			<key>uid</key>
			<string>1A47A645-DB97-46F0-9591-32B0C3A0DD62</string>
			<key>version</key>
			<integer>3</integer>
		</dict>
		<dict>
			<key>config</key>
			<dict>
				<key>autopaste</key>
				<false/>
				<key>clipboardtext</key>
				<string>{query}</string>
				<key>ignoredynamicplaceholders</key>
				<false/>
				<key>transient</key>
				<false/>
			</dict>
			<key>type</key>
			<string>alfred.workflow.output.clipboard</string>
			<key>uid</key>
			<string>593EE2CA-735D-4E27-946D-0B068037202E</string>
			<key>version</key>
			<integer>3</integer>
		</dict>
	</array>
	<key>readme</key>
	<string><![CDATA[
  {{readme}}
  ]]></string>
	<key>uidata</key>
	<dict>
		<key>09857D2F-1057-4384-8BF6-74F4AD3014E2</key>
		<dict>
			<key>xpos</key>
			<integer>495</integer>
			<key>ypos</key>
			<integer>55</integer>
		</dict>
		<key>1A47A645-DB97-46F0-9591-32B0C3A0DD62</key>
		<dict>
			<key>xpos</key>
			<integer>250</integer>
			<key>ypos</key>
			<integer>170</integer>
		</dict>
		<key>5792546E-9F7A-45BA-8D4B-911714944E49</key>
		<dict>
			<key>colorindex</key>
			<integer>12</integer>
			<key>note</key>
			<string>OneUpdater</string>
			<key>xpos</key>
			<integer>725</integer>
			<key>ypos</key>
			<integer>170</integer>
		</dict>
		<key>593EE2CA-735D-4E27-946D-0B068037202E</key>
		<dict>
			<key>xpos</key>
			<integer>500</integer>
			<key>ypos</key>
			<integer>340</integer>
		</dict>
	</dict>
	<key>version</key>
	<string>{{version}}</string>
	<key>webaddress</key>
	<string>https://github.com/Nauxscript/alfred-wechat-stickers</string>
</dict>
</plist>
