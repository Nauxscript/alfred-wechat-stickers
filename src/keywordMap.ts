import type { Keys } from './stickersData'

type KeywordMap = Record<string, Keys[]>

const keywordMap: KeywordMap = {
  '笑 smile': ['旺柴', '奸笑', '社会社会', '呲牙', '再见', '皱眉', '偷笑', '坏笑', '色', '憨笑', '微笑', '阴险', '愉快', '太阳'],
}

export default keywordMap

