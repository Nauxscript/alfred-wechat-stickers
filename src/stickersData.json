{"data":[{"key":"Emm"},{"key":"OK"},{"key":"亲亲"},{"key":"便便"},{"key":"偷笑"},{"key":"傲慢"},{"key":"再见"},{"key":"凋谢"},{"key":"加油"},{"key":"勾引"},{"key":"发呆"},{"key":"发怒"},{"key":"发抖"},{"key":"可怜"},{"key":"右哼哼"},{"key":"吃瓜"},{"key":"吐"},{"key":"呲牙"},{"key":"咒骂"},{"key":"咖啡"},{"key":"哇"},{"key":"啤酒"},{"key":"嘘"},{"key":"嘴唇"},{"key":"嘿哈"},{"key":"囧"},{"key":"困"},{"key":"坏笑"},{"key":"大哭"},{"key":"天啊"},{"key":"太阳"},{"key":"奸笑"},{"key":"好的"},{"key":"委屈"},{"key":"害羞"},{"key":"尴尬"},{"key":"弱"},{"key":"强"},{"key":"得意"},{"key":"微笑"},{"key":"心碎"},{"key":"快哭了"},{"key":"悠闲"},{"key":"惊恐"},{"key":"惊讶"},{"key":"愉快"},{"key":"憨笑"},{"key":"打脸"},{"key":"抓狂"},{"key":"抠鼻"},{"key":"抱拳"},{"key":"拥抱"},{"key":"拳头"},{"key":"捂脸"},{"key":"握手"},{"key":"撇嘴"},{"key":"擦汗"},{"key":"敲打"},{"key":"旺柴"},{"key":"晕"},{"key":"月亮"},{"key":"机智"},{"key":"汗"},{"key":"流泪"},{"key":"炸弹"},{"key":"爱心"},{"key":"猪头"},{"key":"玫瑰"},{"key":"疑问"},{"key":"發"},{"key":"白眼"},{"key":"皱眉"},{"key":"睡"},{"key":"礼物"},{"key":"社会社会"},{"key":"福"},{"key":"红包"},{"key":"胜利"},{"key":"色"},{"key":"菜刀"},{"key":"蛋糕"},{"key":"衰"},{"key":"调皮"},{"key":"跳跳"},{"key":"转圈"},{"key":"鄙视"},{"key":"闭嘴"},{"key":"阴险"},{"key":"难过"},{"key":"骷髅"},{"key":"鼓掌"}]}