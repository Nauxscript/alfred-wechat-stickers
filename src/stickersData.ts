export type Keys = 'Emm' | 'OK' | '亲亲-qq' | '便便-bb' | '偷笑-tx' | '傲慢-am' | '再见-zj' | '凋谢-dx' | '加油-jy' | '勾引-gy' | '发呆-fd' | '发怒-fn' | '发抖-fd' | '可怜-kl' | '右哼哼-yhh' | '吃瓜-cg' | '吐-tu' | '呲牙-zy' | '咒骂-zm' | '咖啡-kf' | '哇-wa' | '啤酒-pj' | '嘘-xu' | '嘴唇-zc' | '嘿哈-hh' | '囧-jiong' | '困-kun' | '坏笑-hx' | '大哭-dk' | '天啊-ta' | '太阳-ty' | '奸笑-jx' | '好的-hd' | '委屈-wq' | '害羞-hx' | '尴尬-gg' | '弱-ruo' | '强-qiang' | '得意-dy' | '微笑-wx' | '心碎-xs' | '快哭了-kkl' | '悠闲-yx' | '惊恐-jk' | '惊讶-jy' | '愉快-yk' | '憨笑-hx' | '打脸-dl' | '抓狂-zk' | '抠鼻-kb' | '抱拳-bq' | '拥抱-yb' | '拳头-qt' | '捂脸-wl' | '握手-ws' | '撇嘴-pz' | '擦汗-ch' | '敲打-qd' | '旺柴-wc' | '晕-yun' | '月亮-yl' | '机智-jz' | '汗-han' | '流泪-ll' | '炸弹-zd' | '爱心-ax' | '猪头-zt' | '玫瑰-mg' | '疑问-yw' | '發-fa' | '白眼-by' | '皱眉-zm' | '睡-shui' | '礼物-lw' | '社会社会-shsh' | '福-fu' | '红包-hb' | '胜利-sl' | '色-se' | '菜刀-cd' | '蛋糕-dg' | '衰-shuai' | '调皮-tp' | '跳跳-tt' | '转圈-zq' | '鄙视-bs' | '闭嘴-bz' | '阴险-yx' | '难过-ng' | '骷髅-kl' | '鼓掌-gz'export default ['Emm', 'OK', '亲亲-qq', '便便-bb', '偷笑-tx', '傲慢-am', '再见-zj', '凋谢-dx', '加油-jy', '勾引-gy', '发呆-fd', '发怒-fn', '发抖-fd', '可怜-kl', '右哼哼-yhh', '吃瓜-cg', '吐-tu', '呲牙-zy', '咒骂-zm', '咖啡-kf', '哇-wa', '啤酒-pj', '嘘-xu', '嘴唇-zc', '嘿哈-hh', '囧-jiong', '困-kun', '坏笑-hx', '大哭-dk', '天啊-ta', '太阳-ty', '奸笑-jx', '好的-hd', '委屈-wq', '害羞-hx', '尴尬-gg', '弱-ruo', '强-qiang', '得意-dy', '微笑-wx', '心碎-xs', '快哭了-kkl', '悠闲-yx', '惊恐-jk', '惊讶-jy', '愉快-yk', '憨笑-hx', '打脸-dl', '抓狂-zk', '抠鼻-kb', '抱拳-bq', '拥抱-yb', '拳头-qt', '捂脸-wl', '握手-ws', '撇嘴-pz', '擦汗-ch', '敲打-qd', '旺柴-wc', '晕-yun', '月亮-yl', '机智-jz', '汗-han', '流泪-ll', '炸弹-zd', '爱心-ax', '猪头-zt', '玫瑰-mg', '疑问-yw', '發-fa', '白眼-by', '皱眉-zm', '睡-shui', '礼物-lw', '社会社会-shsh', '福-fu', '红包-hb', '胜利-sl', '色-se', '菜刀-cd', '蛋糕-dg', '衰-shuai', '调皮-tp', '跳跳-tt', '转圈-zq', '鄙视-bs', '闭嘴-bz', '阴险-yx', '难过-ng', '骷髅-kl', '鼓掌-gz'] as Keys[]