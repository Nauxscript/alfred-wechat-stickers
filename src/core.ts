import { Application } from '@jxa/types'
import type { AlfredSelection } from './index.d'
import keywordMap from './keywordMap'
// import stickersData from './stickersData.json'
import stickersData from './stickersData'
import type { Keys } from './stickersData'
const iconPath = './icons'
// const allStickersData = JSON.parse(JSON.stringify(stickersData.data)) as Sticker[]

export const parseData = (stickers: Keys[], iconPath = './icons'): AlfredSelection[] => {
  return stickers.map((key) => {
    const name = key.split('-')[0]
    return {
      uid: name,
      title: name,
      subtitle: `粘贴“[${name}]”到当前应用中`,
      arg: `[${name}]`,
      icon: {
        path: `${iconPath}/${key}.png`,
      },
      mods: {
        ctrl: {
          arg: `[${name}]`,
          subtitle: `复制“[${name}]”到剪切板中`,
        },
      },
    }
  })
}

export const searchWithKeyword = (query: string) => {
  const keywordRes = Object.keys(keywordMap).reduce((prev, curr) => {
    if (curr.includes(query))
      return [...prev, ...keywordMap[curr]]
    return prev
  }, [] as Keys[])

  const keyRes = stickersData.filter(key => key.includes(query))
  const result = [...new Set([...keywordRes, ...keyRes])] as Keys[]
  return result
}

export const searchFn = (query: string) => {
  if (!query)
    return parseData(stickersData, iconPath)
  const matches = searchWithKeyword(query)
  return parseData(matches, iconPath)
}
