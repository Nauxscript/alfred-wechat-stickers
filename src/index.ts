// const data = await alfy.fetch('https://jsonplaceholder.typicode.com/posts');

import { searchFn } from './core'

export default function run(argv: Array<string | boolean >) {
  const query = argv[0]
  const items = searchFn(query as string)
  const result = { items }
  if (argv[1]) {
    // eslint-disable-next-line no-console
    return result
  }
  // eslint-disable-next-line no-console
  console.log(JSON.stringify(result, null, ''))
}
