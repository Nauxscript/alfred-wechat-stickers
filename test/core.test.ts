import { describe, expect, it } from 'vitest'
import { parseData } from '../src/core'

describe('core test', () => {
  it('parseData function is ok', () => {
    const result = parseData([{ key: '旺柴' }], './icons')
    expect(result).toMatchSnapshot()
  })
})
