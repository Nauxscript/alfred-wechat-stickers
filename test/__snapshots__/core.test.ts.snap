// Vitest Snapshot v1

exports[`core test > parseData function is ok 1`] = `
[
  {
    "arg": "[旺柴]",
    "icon": {
      "path": "./icons/旺柴.png",
    },
    "mods": {
      "ctrl": {
        "arg": "[旺柴]",
        "subtitle": "Copy and paste",
      },
    },
    "subtitle": "旺柴",
    "title": "旺柴",
  },
]
`;
